### ISSUE TITLE
assignment within `while`

### ISSUE THREAD

**Issue by zanona** [douglascrockford/JSLint#40]
when using [couchdb](http://wiki.apache.org/couchdb/Formatting_with_Show_and_List#Listing_Views_with_CouchDB_0.10_and_later) there's a loop which pass through every record in the database, but it only works with the following code:

``` js
var row;
while (row = getRow()) {

  //print some results
  send(row.title);
  send(row.author);

}
```

but this returns the following error on jslint: 

> Expected a conditional expression and instead saw an assignment

The problem is that there's no other way to achieve the same result besides assign a value to `row` from within the `while` argument for example

``` js
//doesn't work
var row = getRow();
while (row) {
  . . .
}

// doesn't work too
var row;
while (getRow()) {
  row = getRow();
  . . .
}

```

Bellow you can see how the `getRow()` method works...

``` js

function getRow() {
    if (lastRow) {
        return null;
    }
    if (!gotRow) {
        gotRow = true;
        sendStart();
    } else {
        blowChunks();
    }
    var line = readline();
    var json = eval("(" + line + ")");
    if (json[0] == "list_end") {
        lastRow = true;
        return null;
    }
    if (json[0] != "list_row") {
        throw ["fatal", "list_error", "not a row '" + json[0] + "'"];
    }
    return json[1];
}
```

So In this case I am just curious about how to get rid of the warning doing it right way, because I couldn't achieve the same result with any alternative.
